src/placeos-core/module_manager.cr

```diff
@@ -81,8 +81,10 @@ module PlaceOS
         when RethinkORM::Changefeed::Event::Deleted
           remove_module(mod)
         when RethinkORM::Changefeed::Event::Updated
-          # Running state of the module has changed
-          if mod.running_changed? && discovery.own_node?(mod.id.as(String))
+          if ModuleManager.needs_restart?(mod)
+            mod.running ? restart_module(mod) : stop_module(mod)
+          elsif mod.running_changed? && discovery.own_node?(mod.id.as(String))
+            # Running state of the module has changed
             mod.running ? start_module(mod) : stop_module(mod)
           end
```

Comment:
Should the `discovery.own_node?` check apply to both of these branches?

Reply:
nice catch!